biogeochem/EDCohortDynamicsMod.F90

```diff
@@ -822,26 +825,28 @@ subroutine fuse_cohorts(currentSite, currentPatch, bc_in)
                                 currentCohort%laimemory   = (currentCohort%n*currentCohort%laimemory   &
                                       + nextc%n*nextc%laimemory)/newn
 
-                                currentCohort%dbh         = (currentCohort%n*currentCohort%dbh         &
-                                      + nextc%n*nextc%dbh)/newn
-
-                                call h_allom(currentCohort%dbh,currentCohort%pft,currentCohort%hite)
-                                
                                 currentCohort%canopy_trim = (currentCohort%n*currentCohort%canopy_trim &
                                       + nextc%n*nextc%canopy_trim)/newn
 
-                                ! -----------------------------------------------------------------
-                                ! If fusion pushed structural biomass to be larger than
-                                ! the allometric target value derived by diameter, we
-                                ! then increase diameter and height until the allometric 
-                                ! target matches actual bdead. (if it is the other way around
-                                ! we then just let the carbon pools grow to fill-out allometry)
-                                ! -----------------------------------------------------------------
+                                ! conserve total crown area during the fusion step, and then calculate dbh of the
+                                ! fused cohort as that which conserves both crown area and the dbh to crown area allometry.  
+                                ! dbh will be updated in the next growth step in the (likely) event that dbh to structural 
+                                ! biomass allometry is exceeded. if using a capped crown area allometry and above the cap, 
+                                ! then calculate as the weighted average of fusing cohorts' dbh
+
+                                currentCohort%c_area = currentCohort%c_area + nextc%c_area
+
+                                call carea_allom(dbh,newn,currentSite%spread,currentCohort%pft,&
+                                     currentCohort%c_area,inverse=.true.)
                                 
-                                if( EDPftvarcon_inst%woody(currentCohort%pft) == itrue ) then
-                                   call StructureResetOfDH( currentCohort%prt%GetState(struct_organ,all_carbon_elements), currentCohort%pft, &
-                                         currentCohort%canopy_trim, currentCohort%dbh, currentCohort%hite )
-                                end if
+                                if (dbh .eq. fates_unset_r8) then
```

Comment:
what is fates_unset_r8?

Reply:
ah, i see, it means it's not set. Fair enough. Maybe add a comment so one doesn't need to go and look up the definition to understand this bit?